core/arch/arm/kernel/boot.c

```diff
@@ -118,30 +118,35 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
-#if defined(_CFG_CORE_STACK_PROTECTOR)
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
 /* Generate random stack canary value on boot up */
-__weak uintptr_t plat_get_random_stack_canary(void)
+__weak void plat_get_random_stack_canaries(void *buf, size_t ncan, size_t size)
 {
-	uintptr_t canary = 0xbaaaad00;
 	TEE_Result ret = TEE_ERROR_GENERIC;
+	size_t i = 0;
+
+	assert(buf && ncan && size);
 
 	/*
 	 * With virtualization the RNG is not initialized in Nexus core.
 	 * Need to override with platform specific implementation.
 	 */
 	if (IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
 		IMSG("WARNING: Using fixed value for stack canary");
-		return canary;
+		memset(buf, 0xab, ncan * size);
+		goto out;
 	}
 
-	ret = crypto_rng_read(&canary, sizeof(canary));
+	ret = crypto_rng_read(buf, ncan * size);
 	if (ret != TEE_SUCCESS)
 		panic("Failed to generate random stack canary");
 
+out:
 	/* Leave null byte in canary to prevent string base exploit */
-	return canary & ~0xffUL;
+	for (i = 0; i < ncan; i++)
+		*((uint8_t *)buf + size * i) &= ~0xffUL;
```

Comment:
Can replace `UL` with `U`, but even prefer `~U(0xff)` altough I think `~0xff` is enough.

Reply:
I attempted to use `~0xff`, but the compiler generated a warning about overflow.

>   CC      out/arm-plat-mediatek/core/arch/arm/kernel/boot.o
core/arch/arm/kernel/boot.c: In function ‘plat_get_random_stack_canaries’:
core/arch/arm/kernel/boot.c:147:35: warning: overflow in implicit constant conversion [-Woverflow]
   *((uint8_t *)buf + size * i) &= ~0xff;
                                   ^